lib/velum/salt_api.rb

```diff
@@ -95,10 +95,8 @@ def perform_request(endpoint:, method:, data: {}, client: nil)
 
         opts = { use_ssl: false, open_timeout: 2 }
         Net::HTTP.start(uri.hostname, uri.port, opts) { |http| http.request(req) }
-      rescue *HTTP_EXCEPTIONS
-        # TODO: we might want to wrap the original exception inside the SaltConnectionException
-        # so we can show detailed problems to the user in the future.
-        raise SaltConnectionException
+      rescue *HTTP_EXCEPTIONS => e
+        raise SaltConnectionException, e
```

Comment:
to wrap the exception details into `SaltConnectionException`, for not getting lost as they will bubble up in the json response or flash message i think